okhttp/src/test/java/okhttp3/HttpUrlTest.java

```diff
@@ -1756,6 +1756,9 @@ HttpUrl parse(String url) {
     assertThat(parse("https://xn--4pvxs.jp").topPrivateDomain()).isNull();
     assertThat(parse("https://localhost").topPrivateDomain()).isNull();
     assertThat(parse("https://127.0.0.1").topPrivateDomain()).isNull();
+
+    // https://github.com/square/okhttp/issues/6109
+    assertThat(parse("http://a./").topPrivateDomain()).isNull();
```

Comment:
Also test these?
```
a..
..a
a..b
.a
.
..

```

Reply:
My interpretation is that hostname ending with '.' is the one special case since it's actually the blessed form of dns, and we mostly just all agreed to drop that trailing '.'

Or is there some special behaviour we want for these?

Reply:
I really don’t know. I was mostly worried that we’d return `squareup.com` as the top private domain, which seemed wrong. But I’d forgotten about the trailing `.` thing in DNS.

Now I’m curious, What Does Guava Do?

Reply:
The awkward thing with this change is that HttpUrl parsing is our shield, but also known to be non exact.  And as much as it doesn't match URI, it will be painful and cause bugs in various places e.g. calling through to JDK methods like ProxySelector.